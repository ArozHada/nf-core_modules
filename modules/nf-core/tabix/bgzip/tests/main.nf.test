nextflow_process {

    name "Test Process TABIX_BGZIP"
    script "modules/nf-core/tabix/bgzip/main.nf"
    process "TABIX_BGZIP"

    test("sarscov2_vcf_bgzip_compress") {

        when {
            process {
                """
                input[0] = [
                                [ id:'test' ],
                                [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("homo_genome_bedgz_compress") {
        when {
            process {
                """
                input[0] = [
                            [ id:'test' ],
                            [ file(params.test_data['homo_sapiens']['genome']['genome_bed_gz'], checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("sarscov2_vcf_bgzip_compress_gzi") {
        config "./bgzip_compress.config"
        when {
            process {
                """
                input[0] = [
                                [ id:'test' ],
                                [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("sarscov2_vcf_bgzip_compress_stub") {
        config "./vcf_none.config"
        options '-stub'
        when {
            process {
                """
                input[0] = [
                                [ id:'test' ],
                                [ file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
