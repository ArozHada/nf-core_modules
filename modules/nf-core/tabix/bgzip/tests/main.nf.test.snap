{
    "homo_genome_bedgz_compress": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,87a15eb9c2ff20ccd5cd8735a28708f7"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,a48a4d4bb9ea4b11b878ce36de0c5163"
                ],
                "gzi": [
                    
                ],
                "output": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bed:md5,87a15eb9c2ff20ccd5cd8735a28708f7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a48a4d4bb9ea4b11b878ce36de0c5163"
                ]
            }
        ],
        "timestamp": "2024-02-16T14:14:00.968546"
    },
    "sarscov2_vcf_bgzip_compress": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,8e722884ffb75155212a3fc053918766"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,a48a4d4bb9ea4b11b878ce36de0c5163"
                ],
                "gzi": [
                    
                ],
                "output": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,8e722884ffb75155212a3fc053918766"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a48a4d4bb9ea4b11b878ce36de0c5163"
                ]
            }
        ],
        "timestamp": "2024-02-16T14:13:51.710694"
    },
    "sarscov2_vcf_bgzip_compress_gzi": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,8e722884ffb75155212a3fc053918766"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.gzi:md5,7dea362b3fac8e00956a4952a3d4f474"
                    ]
                ],
                "2": [
                    "versions.yml:md5,a48a4d4bb9ea4b11b878ce36de0c5163"
                ],
                "gzi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.gzi:md5,7dea362b3fac8e00956a4952a3d4f474"
                    ]
                ],
                "output": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,8e722884ffb75155212a3fc053918766"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a48a4d4bb9ea4b11b878ce36de0c5163"
                ]
            }
        ],
        "timestamp": "2024-02-16T14:14:10.253424"
    },
    "sarscov2_vcf_bgzip_compress_stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "/nfs/treeoflife-01/teams/tola/users/dp24/new/modules/.nf-test/tests/456a0def45e880cce9d6bc7218aaa66/work/71/22a0fc7ed4435a13d2da59baa56e4a/test.vcf.gz"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,a48a4d4bb9ea4b11b878ce36de0c5163"
                ],
                "gzi": [
                    
                ],
                "output": [
                    [
                        {
                            "id": "test"
                        },
                        "/nfs/treeoflife-01/teams/tola/users/dp24/new/modules/.nf-test/tests/456a0def45e880cce9d6bc7218aaa66/work/71/22a0fc7ed4435a13d2da59baa56e4a/test.vcf.gz"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a48a4d4bb9ea4b11b878ce36de0c5163"
                ]
            }
        ],
        "timestamp": "2024-02-16T14:14:19.49874"
    }
}